src/main/java/org/openrewrite/java/spring/swagger/TransformApiInfo.java

```diff
@@ -0,0 +1,155 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.spring.swagger;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.*;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+public class TransformApiInfo extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Transform SpringFox `ApiInfo` to Swagger v3 `Info`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Transforms SpringFox `ApiInfoBuilder` to Swagger v3 `Info` fluent API pattern.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(
+                new UsesType<>("springfox.documentation.builders.ApiInfoBuilder", true),
+                new TransformApiInfoVisitor()
+        );
+    }
+
+    private static class TransformApiInfoVisitor extends JavaIsoVisitor<ExecutionContext> {
+        private final MethodMatcher apiInfoBuilderConstructor = new MethodMatcher("springfox.documentation.builders.ApiInfoBuilder <constructor>()");
+        private final MethodMatcher termsOfServiceUrl = new MethodMatcher("springfox.documentation.builders.ApiInfoBuilder termsOfServiceUrl(String)");
+        private final MethodMatcher licenseUrl = new MethodMatcher("springfox.documentation.builders.ApiInfoBuilder licenseUrl(String)");
+        private final MethodMatcher license = new MethodMatcher("springfox.documentation.builders.ApiInfoBuilder license(String)");
+        private final MethodMatcher contact = new MethodMatcher("springfox.documentation.builders.ApiInfoBuilder contact(springfox.documentation.service.Contact)");
+        private final MethodMatcher contactConstructor = new MethodMatcher("springfox.documentation.service.Contact <constructor>(String, String, String)");
+        private final MethodMatcher build = new MethodMatcher("springfox.documentation.builders.ApiInfoBuilder build()");
+
+        // accumulator for AppInfoBuilder methods
+        private static class Accumulator {
+            private final Map<String, Function<J.MethodInvocation, String>> transformations = new HashMap<>();
```

Comment:
A map of name -> lambda is a lot like a class with named methods, just less efficient.